src/Todo.jsx

```diff
@@ -0,0 +1,36 @@
+import React, { useState } from 'react';
+import * as R from 'ramda';
+import List from './List';
+import Input from './Input';
+
+export default function Todo() {
+  const [userInput, setUserInput] = useState('');
+  const [tasks, setTasks] = useState([]);
+
+  function updatInputText(e) {
+    setUserInput(e.target.value);
+  }
+
+  function deleteTask(target) {
+    const isTaskToRemain = R.pipe(R.prop('id'), R.equals(target), R.not);
+    setTasks((oldTasks) => (R.filter(isTaskToRemain, oldTasks)));
+  }
+
+  function addTask() {
+    const newId = R.last(R.map(R.prop('id'), tasks)) + 1 || 0;
+    const newContent = userInput;
+
+    setTasks((oldTasks) => (
+      [...oldTasks, { id: newId, content: newContent }]
+    ));
+    document.getElementById('inputBox').value = '';
+  }
+
+  return (
+    <p>
+      <h1>To-do</h1>
+      <Input updater={updatInputText} submitter={addTask} />
```

Comment:
```suggestion
      <Input onChangeInput={handleChangeInput} onSubmit={handleAddTask} />
```

위 예시와 같은 네이밍 컨벤션은 어떤가요? 실제 일을 처리하는 관점과 사용하는 입장에서 바라볼 때의 의도를 각각 드러내줄 수 있을 것 같아요.

참고:
- https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364

Reply:
굉장히 좋은 말씀입니다. `3ffc815` `16055c1`으로 수정했습니다.

```jsx
<p>
  <h1>To-do</h1>
  <Input onChangeInput={handleChangeInput} onSubmit={handleAddTask} />
  <List items={tasks} onClick={handleDeleteTask} />
</p>
```
이것만 봐도 무엇을 하는지를 명확하게 읽히네요.